### DISCUSSION TITLE
Controlling Incline & Resistance from the app

### DISCUSSION THREAD

**Discussion by rjlee** [cagnulein/qdomyos-zwift#2252] [Q&A] [answered]
I've successfully got the app running on my iPhone talking to my Nordic Track S22i bike, everything is reporting correctly (including seeing changes to incline and resistance when I change them manually on the bike).  However, when I click the +/- button on the tile for either of these, nothing happens.  Is there something I need to configure to enable control of these from the app ?

**Comment by cagnulein** [comment]
hi! did you install the companion via adb?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 24 mar 2024 alle 21:33 Robert Lee ***@***.***>
ha scritto:

**Reply by rjlee** [reply, marked as answer]
I've done it a couple of ways, I *think* the last time I installed it was via the tinyurl link.  I've just done an `adb install` and now incline/decline appears to work, however resistance doesn't respond to pressing +/- unfortunately.  Any ideas on that one ?

**Comment by cagnulein** [comment]
yes because i did in this way. incline also change the resistance on the
bike, so 2 birds with 1 bullet :)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 24 mar 2024 alle 21:45 Robert Lee ***@***.***>
ha scritto:

**Comment by rjlee** [comment]
Ok, thanks.  

I've also got the QZ app running on the S22i console itself as well, (installed via play store which I've installed on the S22i Android 9 console).  That appears to report the metrics fine, but the incline/decline +/- don't work on the app installed on the console.

Is there any issue with running the companion app and the QZ app both together on the console itself (so same device) ?

**Comment by cagnulein** [comment]
wow this sounds amazing! never try this!
could you please show me a debug log from this scenario?
https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 24 mar 2024 alle 22:02 Robert Lee ***@***.***>
ha scritto:

**Comment by rjlee** [comment]
I've just been doing some testing with Zwift and in some instances, Zwift does appear to be able to control the incline/decline.  I'm wondering if there is some kind of backgroud/foreground app issue, but I'll do some further testing and then submit an issue.

Just so you know what I'm trying to achieve:

1. Screen mirroring on my iphone to cast my screen to the S22i tablet - this works - there is a little bit of upscaling, but casting works pretty well.  This obviously means you can use the main console to display zwift
2. Run the companion app and the qz app on the S22i console, so I don't need any other devices.  This works well for Watts/Cadence (whilst running a manual iFit workout in the background), but in the qz app the incline controls don't appear to function - again going to do some more testing on this.

However, it's pretty good already, with just my iPhone and the S22i I can use Zwift and the cadence/rpm metrics all work fine - I'm just trying to understand under what situations the incline (and resistance) do and don't work.

**Reply by cagnulein** [reply]
@rjlee the auto inclination works only if you have the manual ifit view on foreground. Maybe that's your issue?
@victorypoint maybe you're interesting in this too!

**Comment by rjlee** [comment]
From testing, I can confirm that screen mirroring to a casting app (AirCast) on the main console means that incline/decline doesn't work (but when not casting to the screen it does), so it looks like it may be related to the foreground/background app issue.  I can grab some logs if you'd like to have a look, let me know.

**Reply by cagnulein** [reply]
Yes as I said qz is using the virtual touch on ifit to change the resistance, so ifit must be running on full screen for sure for auto resistance
